src/resource/Resource.ts

```diff
@@ -306,11 +306,11 @@ export default abstract class Resource {
       type: 'mutate',
       schema: self.getEntitySchema(),
       options,
-      getUrl(params: object) {
-        return self.url(params);
+      getFetchKey(params: object) {
+        return 'PUT ' + self.url(params);
       },
-      fetch(url: string, body: Partial<AbstractInstanceType<T>>) {
-        return self.fetch('put', url, body);
+      fetch(params: Readonly<object>, body: Partial<AbstractInstanceType<T>>) {
+        return self.fetch('put', self.url(params), body);
```

Comment:
nit: why is PUT uppercase in getFetchKey but lowercase in self.fetch?

Reply:
lowercase because it's used for method name in superagent....upper case is kinda arbitrary but i guess it seems to call it out more distinctly